src/httomolib/prep/stripe.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ---------------------------------------------------------------------------
+# Copyright 2022 Diamond Light Source Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ---------------------------------------------------------------------------
+# Created By  : Tomography Team at DLS <scientificsoftware@diamond.ac.uk>
+# Created Date: 01 November 2022
+# version ='0.1'
+# ---------------------------------------------------------------------------
+"""Modules for stripes removal using CuPy API"""
+import cupy as cp
+from cupy import abs, mean, ndarray
+from cupyx.scipy.ndimage import median_filter
+
+
+def remove_stripes_titarenko_cupy(data: ndarray,
+                             beta: float = 0.1) -> ndarray:
+    """
+    Removes stripes with the method of V. Titarenko (TomoCuPy implementation)
+
+    Parameters
+    ----------
+    arr : ndarray
+        3D stack of projections as a CuPy array.
+    beta : float, optional
+        filter parameter, lower values increase the filter strength.
+
+    Returns
+    -------
+    ndarray
+        3D CuPy array of de-striped projections.
+    """
+    gamma = beta * ((1 - beta) / (1 + beta)) ** abs(
+        cp.fft.fftfreq(data.shape[-1]) * data.shape[-1]
+    )
+    gamma[0] -= 1
+    v = mean(data, axis=0)
+    v = v - v[:, 0:1]
+    v = cp.fft.irfft(cp.fft.rfft(v) * cp.fft.rfft(gamma))
+    data[:] += v
+
+    return data
+
+
+# Naive CuPy port of the NumPy implementation in TomoPy
+def remove_stripe_based_sorting_cupy(tomo: ndarray,
```

Comment:
The GPU method names should be the same as in [TomoPy](https://github.com/tomopy/tomopy/blob/9ccedc67f7e6f023ca1057c883f60167ac5bd647/source/tomopy/prep/stripe.py#L79), right?
```suggestion
def remove_stripe_based_sorting(tomo: ndarray,
```
Also, we need `__all__` to list the user-facing GPU methods.

Reply:
I'm having second thoughts that it should be exactly as in TomoPy. I think it could be useful if we have a _gpu_ identifier at the end of the name to differentiate better as there will be CPU methods as well. E.g. _remove_stripe_based_sorting -> remove_stripe_based_sorting_gpu_
Agree with the `__all__` thing.